src/platform/packages/private/kbn-esql-editor/src/helpers.ts

```diff
@@ -63,15 +63,19 @@ export const useDebounceWithOptions = (
   );
 };
 
-const quotedWarningMessageRegexp = /"(.*?)"/g;
+// Quotes can be used as separators for multiple warnings unless
+// they are escaped with backslashes. This regexp will match any
+// quoted string that is not escaped.
+const quotedWarningMessageRegexp = /"(?:[^"\\]*(?:\\.[^"\\]*)*)"/g;
 
 export const parseWarning = (warning: string): MonacoMessage[] => {
   if (quotedWarningMessageRegexp.test(warning)) {
     const matches = warning.match(quotedWarningMessageRegexp);
     if (matches) {
       return matches.map((message) => {
         // start extracting the quoted message and with few default positioning
-        let warningMessage = message.replace(/"/g, '');
+        // replaces the quotes only if they are not escaped
+        let warningMessage = message.replace(/(?<!\\)"/g, '').replace(/\\/g, '');
```

Comment:
nit: can be a single `replace` to avoid double traversal of a string 
```suggestion
        let warningMessage = message.replace(/(?<!\\)"|\\/g, '');
```